manifests/osfamily/redhat.pp

```diff
@@ -88,14 +88,16 @@
     logoutput => 'on_failure',
   }
 
-  yumrepo { 'pc_repo':
-    baseurl       => $source,
-    descr         => "Puppet Labs ${::puppet_agent::collection} Repository",
-    enabled       => true,
-    gpgcheck      => '1',
-    gpgkey        => "${gpg_keys}",
-    sslcacert     => $_sslcacert_path,
-    sslclientcert => $_sslclientcert_path,
-    sslclientkey  => $_sslclientkey_path,
+  if $::puppet_agent::manage_repo {
```

Comment:
Works for me, putting the gpg keys there shouldn't prevent any other configuration.